src/main/java/org/springframework/samples/petclinic/rest/importcsv/ImportCSV.java

```diff
@@ -33,106 +136,62 @@
         produces = "application/json")
     public ResponseEntity<List<Pet>> importPets(@RequestBody String csv) {
 
-        int i = 0;
+        int pos = 0;
```

Comment:
this method is maybe still a bit long...maybe you could extract some methods to create e.g., the `importPets` method only handles the http request, and then you have a new methods that orchestrates the parsing and then you have many small methods that to the parsing of the various fields and updating the database